### PR TITLE
update error-prone, patterns, and compiler config

### PR DESCRIPTION
**Pull request by jpdahlke**
Updates error-prone and sets us up to hopefully clear the issues and activate this by default to fail builds.

Here are the outstanding issues needing attentions based on this PR:
```
     23 [MemberName]
     13 [SystemOut]
     10 [ReturnMissingNullable]
      8 [FieldMissingNullable]
      5 [YodaCondition]
      4 [NonFinalStaticField]
      4 [MissingSummary]
      4 [ConstantField]
      4 [BooleanParameter]
      3 [CatchingUnchecked]
      2 [UnnecessarilyFullyQualified]
      2 [UngroupedOverloads]
      2 [ThreadPriorityCheck]
      2 [ParameterMissingNullable]
      2 [NotJavadoc]
      2 [CheckedExceptionNotThrown]
      2 [AvoidObjectArrays]
      1 [UnnecessaryParentheses]
      1 [ReturnsNullCollection]
      1 [MissingOverride]
      1 [MethodCanBeStatic]
      1 [JdkObsolete]
      1 [Interruption]
      1 [FieldCanBeFinal]
      1 [EqualsGetClass]
```

### CODE REVIEW

**Comment by jpdahlke** [comment]
- [ ] available internally

File: pom.xml [hunk 1]
```diff
@@ -54,7 +54,7 @@
     <dep.commons-lang.version>3.12.0</dep.commons-lang.version>
     <dep.commons-pool2.version>2.11.1</dep.commons-pool2.version>
     <dep.dropwizard.metrics.version>4.2.4</dep.dropwizard.metrics.version>
-    <dep.errorprone.version>2.10.0</dep.errorprone.version>
+    <dep.errorprone.version>2.24.1</dep.errorprone.version>
```
Comment (jpdahlke) [inline pom.xml:57]:
```suggestion
    <dep.errorprone.version>2.23.0</dep.errorprone.version>
```

**Review by sambish5 — APPROVED** [review]
LGTM

**Review by fbruton — APPROVED** [review]

**Comment by arp-0984** [comment]
Tackling this chunk:
      13 [SystemOut]
     10 [ReturnMissingNullable]
      8 [FieldMissingNullable]
      5 [YodaCondition]
      4 [NonFinalStaticField]
      4 [MissingSummary]
      4 [ConstantField]

**Pull request merged**